conformance/lang/expressions/logical-expr/logical_or_expr.balt

```diff
@@ -0,0 +1,853 @@
+    io:println(2 === 3 || 3 === 3); // @output true
+
+    io:println(true || 2 !== 3); // @output true
+    io:println(3 !== 3 || true); // @output true
+    io:println(false || 3 !== 3); // @output false
+    io:println(2 !== 3 || false); // @output true
+    io:println(2 !== 3 || 3 !== 3); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression with logical expression.
+Labels: logical-expr
+
+function init() {
+    io:println(true || true || false); // @output true
+    io:println(true || false || true); // @output true
+    io:println(false || true || false); // @output true
+    io:println(true || false || false); // @output true
+    io:println(true || false || true || false); // @output true
+}
+
+Test-Case: output
+Description: Expression evaluation order when lhs expression evaluates to false and rhs expression evaluates to true in a logical or expression.
+Labels: logical-expr, function-call-expr, additive-expr, is-expr
+
+int falseEval = 0;
+int trueEval = 0;
+
+function returnTrueVal() returns boolean {
+    trueEval = trueEval + 1;
+    return true;
+}
+
+function returnFalseVal() returns boolean {
+    falseEval = falseEval + 1;
+    return false;
+}
+
+function init() {
+    boolean a = returnFalseVal() || returnTrueVal();
+    io:println(falseEval); // @output 1
+    io:println(trueEval); // @output 1
+    io:println(a is true); // @output true
+}
+
+Test-Case: output
+Description: Expression evaluation order when lhs expression evaluates to true and rhs expression evaluates to false in a logical or expression.
+Labels: logical-expr, function-call-expr, additive-expr, is-expr
+
+int falseEval = 0;
+int trueEval = 0;
+
+function returnTrueVal() returns boolean {
+    trueEval = trueEval + 1;
+    return true;
+}
+
+function returnFalseVal() returns boolean {
+    falseEval = falseEval + 1;
+    return false;
+}
+
+function init() {
+    boolean a = returnTrueVal() || returnFalseVal();
+    io:println(falseEval); // @output 0
+    io:println(trueEval); // @output 1
+    io:println(a is true); // @output true
+}
+
+Test-Case: output
+Description: Expression evaluation order when both lhs and rhs expressions evaluate to true in a logical or expression.
+Labels: logical-expr, function-call-expr, additive-expr, is-expr
+
+int falseEval = 0;
+int trueEval = 0;
+
+function returnTrueVal() returns boolean {
+    trueEval = trueEval + 1;
+    return true;
+}
+
+function returnFalseVal() returns boolean {
+    falseEval = falseEval + 1;
+    return false;
+}
+
+function init() {
+    boolean a = returnTrueVal() || returnTrueVal();
+    io:println(trueEval); // @output 1
+    io:println(a is true); // @output true
+}
+
+Test-Case: output
+Description: Expression evaluation order when both lhs and rhs expressions evaluate to false in a logical or expression.
+Labels: logical-expr, function-call-expr, additive-expr, is-expr
+
+int falseEval = 0;
+int trueEval = 0;
+
+function returnTrueVal() returns boolean {
```

Comment:
Not used.

Reply:
[94d5da7](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/94d5da7d6a3c741d8d2a8cb2d22ac6910b20f776)